src/Microsoft.OData.Edm/ExtensionMethods/ExtensionMethods.cs

```diff
@@ -1529,6 +1565,30 @@ public static IEdmTypeDefinitionReference GetUInt64(this EdmModel model, string
                 isNullable);
         }
 
+        /// <summary>
+        /// Get the <see cref="IEdmTargetPath"/> from a target path string.
+        /// </summary>
+        /// <param name="model">The EdmModel.</param>
+        /// <param name="targetPath">Target path containing segments separated by '/'. For example: "A.B/C/D.E/Func1(NS.T,NS.T2)/P1".</param>
+        /// <param name="caseInsensitive">Property name case-insensitive or not.</param>
+        /// <returns>The created <see cref="IEdmTargetPath"/>.</returns>
+        public static IEdmTargetPath GetTargetPath(this IEdmModel model, string targetPath, bool caseInsensitive = true)
+        {
+            EdmUtil.CheckArgumentNull(model, nameof(model));
+            EdmUtil.CheckArgumentNull(targetPath, nameof(targetPath));
+
+            string[] targetPathSegments = targetPath.Split('/');
+
+            IEnumerable<IEdmElement> pathSegments = model.GetTargetSegments(targetPathSegments, caseInsensitive);
+
+            if (pathSegments.Any())
+            {
+                return new EdmTargetPath(pathSegments);

```

Comment:
Calling this constructor will call `GetPathString()`, I wonder if that will generate the same path that was used to retrieve the `EdmTargetPath`. If the paths are the same, then it seems unnecessarily expensive to create a path that has just been computed. Maybe we can have an internal `EdmTargetPath(pathSegments, path)` constructor that accepts the path when it's already known, to avoid computing something that is already known.

Reply:
`internal` constructor is a good idea